src/main/java/net/snowflake/client/core/arrow/StructuredTypeConversionHelper.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+package net.snowflake.client.core.arrow;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.arrow.vector.util.JsonStringHashMap;
+
+class StructuredTypeConversionHelper {
+  private static final Set<String> MAP_FIELDS =
+      new HashSet<String>() {
+        {
+          add("key");
+          add("value");
+        }
+      };
+
+  static Object mapHashMapToObject(List<JsonStringHashMap<String, Object>> entriesList) {
+    if (entriesList == null) {
+      return null;
+    }
+    if (entriesList.stream().allMatch(it -> it.keySet().equals(MAP_FIELDS))) {
```

Comment:
That is very tricky... What if someone defines a list of structs with such fields?

Reply:
That's the open question we discussed before - without metadata here we are not able to decide if we have object or map
@sfc-gh-pmotacki WDYT?

Reply:
At the end of the day we should to transfer those information here.